renku_notebooks/api/classes/k8s_client.py

```diff
@@ -115,6 +120,119 @@ def create_server(self, manifest: Dict[str, Any]) -> Dict[str, Any]:
         )
         return server
 
+    def hibernate_server(self, server_name: str, access_token: str):
```

Comment:
The k8s client should not worry about calling the git sidecar. So we should put this in a separate util function or a separate adapter/class.

I would even go as far as only adding a "patch" command on the k8s client that accepts the patch and server name and lightly wraps the k8s client command. 

The k8s client should only worry about running the right k8s command in the right namespace.

Reply:
The logic for hibernate and resume should go to `classes/server.py` right next to the start method in the `UserServer` class. I know this class is insane but my plan is to refactor this in the future and slim it down to only do start/stop/delete/hibernate/resume and remove all the other checks and stuff in other adapters/classes.

Reply:
As discussed, I moved this to a utility file and do it in the `notebooks` handlers. I've renamed the methods in the k8s client to `patch_server` which only call the patching API.

Reply:
Thank you!